src/uct/ib/rc/accel/rc_mlx5.inl

```diff
@@ -267,13 +328,13 @@ uct_rc_mlx5_iface_tm_common_data(uct_rc_mlx5_iface_common_t *iface,
         /* uct_rc_mlx5_iface_common_data will initialize flags value */
         hdr        = uct_rc_mlx5_iface_common_data(iface, cqe, byte_len, flags);
         *flags    |= UCT_CB_PARAM_FLAG_FIRST;
-        *context_p = &dummy_context;
+        *context_p = &iface->tm.mp.dummy_ctx;
         return hdr;
     }
 
     ucs_assert(byte_len <= UCT_RC_MLX5_MP_RQ_BYTE_CNT_FIELD_MASK);
     *flags     = 0;
-    *context_p = uct_rc_mlx5_iface_rx_mp_context(iface, cqe, flags);
+    *context_p = uct_rc_mlx5_iface_rx_mp_context(iface, cqe, flags, has_ep);
```

Comment:
this function is called once
need to separate to 2 functions, one with ep (Rc) another without ep (dc)
and check has_ep here
need to avoid having 1 function which handles 2 completely different cases with nothing common